File: scaffold/src/main/java/top/xuqingquan/web/x5/IAgentWebSettings.java
package top.xuqingquan.web.x5;

import com.tencent.smtt.sdk.WebSettings;
import com.tencent.smtt.sdk.WebView;

public interface IAgentWebSettings<T extends WebSettings> {

    @SuppressWarnings("UnusedReturnValue")
    IAgentWebSettings toSetting(WebView webView);

    T getWebSettings();

}

File: scaffold/src/main/java/top/xuqingquan/web/x5/IAgentWebSettings.kt
package top.xuqingquan.web.x5

import com.tencent.smtt.sdk.WebSettings
import com.tencent.smtt.sdk.WebView

interface IAgentWebSettings<T : WebSettings> {

    fun getWebSettings(): T?

    fun toSetting(webView: WebView?): IAgentWebSettings<*>

}
